paimon-core/src/main/java/org/apache/paimon/schema/SchemaManager.java

```diff
@@ -146,7 +146,23 @@ public TableSchema createTable(Schema schema) throws Exception {
                 }
                 options.remove(CoreOptions.PRIMARY_KEY.key());
             }
-
+            List<String> finalPrimaryKeys = primaryKeys;
+            fields =
```

Comment:
Actually, I don't get why.

`Schema.normalizeFields` will convert pk fields to not-null.

Reply:
when hive create table primary key  is created by 
![image](https://github.com/apache/incubator-paimon/assets/48040205/106a412e-4743-4ca9-b799-489b36570645)
is not from Schema.normalizeFields

Reply:
I see, maybe we should move the logical from `SchemaManager` to `Schema`.